### PR TITLE
Fix #1438: powerauth-client-model classes are duplicated in powerauth-fido2

### LINKED ISSUE #1438: powerauth-client-model classes are duplicated in powerauth-fido2
For example `AllowCredentials`.

### PR DESCRIPTION
**Pull request by banterCZ**
Quite a big refactor. We may merge it after release. Tested end-to-end via `PowerauthFido2TestApplication`.

- Introduce `fido2-model` module.
- Client and servlet use the same request/response objects. Moreover, it fixes the swagger to describe the real payload.
- Structured data used by internal logic such as `CollectedClientData` stored in wrappers.

### CODE REVIEW

File: powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/enumeration/ActivationStatus.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package com.wultra.security.powerauth.fido2.model.enumeration;
+
+/**
+ * Enum representing activation status.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+public enum ActivationStatus {
```
Comment (banterCZ) [inline powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/enumeration/ActivationStatus.java:26]: 🤔 This duplicates `client-model`.

File: powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/error/PowerAuthClientException.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.io.Serial;
+import java.util.Optional;
+
+/**
+ * PowerAuth client exception.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class PowerAuthClientException extends Exception {
```
Comment (banterCZ) [inline powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/error/PowerAuthClientException.java:28]: 🤔 This duplicates `client-model`.

File: powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/entity/AuthenticatorParameters.java [hunk 3]
```diff
@@ -39,6 +41,7 @@ public class AuthenticatorParameters {
     @NotBlank
     private String type;
     private String authenticatorAttachment;
+    @NotNull
```
Comment (banterCZ) [inline powerauth-fido2-model/src/main/java/com/wultra/security/powerauth/fido2/model/entity/AuthenticatorParameters.java:44]: This expectation is in the implicit validator.

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/rest/model/converter/AssertionVerificationRequestWrapperConverter.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        throw new IllegalStateException("Should not be instantiated");
+    }
+
+    /**
+     * Convert the given request into a wrapper.
+     *
+     * @param source assertion verification request
+     * @return wrapped registration request
+     */
+    public static AssertionVerificationRequestWrapper convert(final AssertionVerificationRequest source) throws Fido2DeserializationException {
```
Comment (romanstrobl) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/rest/model/converter/AssertionVerificationRequestWrapperConverter.java:47]: I suggest we use a `@Component` instead for consistence with other converters in this package.

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/rest/model/converter/RegistrationRequestWrapperConverter.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        throw new IllegalStateException("Should not be instantiated");
+    }
+
+    /**
+     * Convert the given request into a wrapper.
+     *
+     * @param source registration request
+     * @return wrapped registration request
+     */
+    public static RegistrationRequestWrapper convert(final RegistrationRequest source) throws Fido2DeserializationException {
```
Comment (romanstrobl) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/rest/model/converter/RegistrationRequestWrapperConverter.java:47]: I suggest we use a `@Component` instead for consistence with other converters in this package.

**Review by romanstrobl — DISMISSED** [review]
Looks OK, I found small issues.

**Review by romanstrobl — APPROVED** [review]
Great!

**Pull request merged**